core/audits/viewport.js

```diff
@@ -46,15 +46,27 @@ class Viewport extends Audit {
   static async audit(artifacts, context) {
     const viewportMeta = await ViewportMeta.request(artifacts.MetaElements, context);
 
+    /** @type {LH.Audit.Details.Base} */
+    const details = {
```

Comment:
so uh.... maybe metricSavings should have been placed on Audit.Product?

Reply:
That's where #14997 put them. We'll definitely have to figure out the larger story of audit details/opportunities/score at some point. The direction might become clearer as we continue to add these and we start to tweak sorting.

Speaking of which, does it make sense to prefix with an underscore or something to make it clearer that these are in progress and may change without a semver/breaking change warning?